x-pack/plugins/endpoint/public/applications/endpoint/view/policy/policy_forms/protections/malware.tsx

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import styled from 'styled-components';
+import { EuiRadio, EuiSwitch, EuiTitle, EuiSpacer } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { OS, ProtectionModes, MalwareProtectionOSes } from '../../../../types';
+import { ConfigForm } from '../config_form';
+import { policyConfig } from '../../../../store/policy_details/selectors';
+import { usePolicyDetailsSelector } from '../../policy_hooks';
+import { clone } from '../../../../models/policy_details_config';
+
+const ProtectionRadioGroup = styled.div`
+  display: flex;
+  .policyDetailsProtectionRadio {
+    margin-right: ${props => props.theme.eui.euiSizeXXL};
+  }
+`;
+
+export const MalwareProtections = React.memo(() => {
+  const policyDetailsConfig = usePolicyDetailsSelector(policyConfig);
+  const dispatch = useDispatch();
+  const selected = policyDetailsConfig && policyDetailsConfig.windows.malware.mode;
+  const OSes: MalwareProtectionOSes[] = [OS.windows, OS.mac];
+  const protection = 'malware';
+
+  const radios = [
+    {
+      id: 'malware.detect',
+      label: i18n.translate('xpack.endpoint.policy.details.detect', { defaultMessage: 'Detect' }),
+    },
+    {
+      id: 'malware.prevent',
+      label: i18n.translate('xpack.endpoint.policy.details.prevent', { defaultMessage: 'Prevent' }),
+    },
+    {
+      id: 'malware.preventNotify',
+      label: i18n.translate('xpack.endpoint.policy.details.preventAndNotify', {
+        defaultMessage: 'Prevent and notify user',
+      }),
+    },
+  ];
+
+  const handleRadioChange = useCallback(
+    e => {
+      const radioId = e.target.id;
+      const separater = radioId.indexOf('.');
```

Comment:
Can you explain what is going on here?

Reply:
the radio ids should be unique to the protection, so i added the protection name on the id. when we save the actual value for the protection, we don't need the protection identifier, just 'detect', 'prevent' preventNotify' or 'off', so i'm just taking off that identifier here since the radio id name looks something like: protection.detect